app-backend/common/src/main/scala/cache/HeapBackedMemcachedClient.scala

```diff
@@ -0,0 +1,76 @@
+package com.azavea.rf.common.cache
+
+import com.github.blemale.scaffeine.{ Cache => ScaffeineCache, Scaffeine }
+import net.spy.memcached._
+
+import scala.concurrent._
+import scala.concurrent.ExecutionContext
+import java.util.concurrent.Executors
+import scala.concurrent.duration._
+import scala.util.matching._
+
+
+/**
+  * Wraps a MemcachedClient to provide a local cache which stores Future results in case
+  *  the call has recently been made and could otherwise result in a race condition.
+  *
+  *  @note Each instance of this wrapper will have its own instance of a Caffeine
+  *         cache (a guava-inspired java library).
+  */
+class HeapBackedMemcachedClient[CachedType](
+  client: MemcachedClient,
+  options: HeapBackedMemcachedClient.Options = HeapBackedMemcachedClient.Options()) {
+  implicit val ec = options.ec
+
+  /** The caffeine cache (on heap) which prevents race conditions */
+  val onHeapCache: ScaffeineCache[String, Future[CachedType]] =
+    Scaffeine()
+      .recordStats()
+      .expireAfterAccess(options.ttl)
+      .maximumSize(options.maxSize)
+      .build[String, Future[CachedType]]()
+
+  /**
+    * The primary means of interaction with this class: pass as key and
+    *  the costly option caching prevents duplication of.
+    */
+  def caching(cacheKey: String)(expensiveOperation: => Future[CachedType])(implicit ec: ExecutionContext): Future[CachedType] = {
+    val sanitizedKey = HeapBackedMemcachedClient.sanitizeKey(cacheKey)
+    val futureCached: Future[CachedType] =
+      onHeapCache.get(sanitizedKey, { cacheKey: String => client.getOrElseUpdate[CachedType](cacheKey, expensiveOperation, options.ttl) })
+    onHeapCache.put(sanitizedKey, futureCached)
+    futureCached
+  }
+}
+
+object HeapBackedMemcachedClient {
+  /** The ExecutionContext to be used in almost all cases for this cache wrapper */
+  private[cache] val defaultExecutionContext = {
+    /** In lieu of a specified configuration, we'll base this off the number of processors
+      *  and the assumption that the cache will be more I/O than CPU bound (thus requiring
+      *  extra threads).
+      */
+    val processors = Runtime.getRuntime().availableProcessors
+    val threads = processors * 10
```

Comment:
Moved the configs to the `common` package and exposed all parameters of memcached and the default params for the on heap component.